contracts/lazyMint/ILazyMintERC1155.sol

```diff
@@ -0,0 +1,124 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity ^0.8.0;
+
+/**
+ *  `LazyMintERC1155` is an ERC 1155 contract. It takes in a base URI in its
+ *  constructor (e.g. "ipsf://Qmece.../"), and the URI for each token of ID
+ *  `tokenId` is baseURI + `${tokenId}` (e.g. "ipsf://Qmece.../1").
+ *
+ *  For each token with a unique ID, the module admin (account with `DEFAULT_ADMIN ROLE`)
+ *  can create mint conditions with non-overlapping time windows, and accounts can claim
+ *  the NFTs, in a given time window, according to that time window's mint conditions.
+ */
+
+interface ILazyMintERC1155 {
+
+    /**
+     *  @notice The mint conditions for a given tokenId x time window.
+     *
+     *  @param startTimestamp The unix timestamp after which the mint conditions last.
+     *                        The same mint conditions last until the `startTimestamp`
+     *                        of the next mint condition.
+     *
+     *  @param maxMintSupply The maximum number of tokens of the same `tokenId` that can
+     *                       be claimed under the mint condition.
+     *
+     *  @param currentMintSupply At any given point, the number of tokens of the same `tokenId`
+     *                           that have been claimed.
+     *
+     *  @param quantityLimitPerTransaction The maximum number of tokens a single account can
+     *                                     claim in a single transaction.
+     *
+     *  @param waitTimeInSecondsBetweenClaims The least number of seconds an account must wait
+     *                                        after claiming tokens, to be able to claim again.
+     *
+     *  @param merkleRoot Only accounts whose address is a leaf of `merkleRoot` can claim tokens
+     *                    under the mint condition.
+     *
+     *  @param pricePerToken The price per token that can be claimed.
+     *
+     *  @param currency The currency in which `pricePerToken` must be paid.
+     */
+    struct MintCondition {
+        uint256 startTimestamp;
+        uint256 maxMintSupply;
+        uint256 currentMintSupply;
+        uint256 quantityLimitPerTransaction;
+        uint256 waitTimeInSecondsBetweenClaims;
```

Comment:
Let's do mint -> claim. I didn't do this before to maintain some backward compatibility, but I'll change names (min -> claim wherever it makes sense) since the SDK team has requested renaming as well.

Reply:
See ca84893dbf2815c2f440253e3df741df3ed128e6